enterprise/api/v1/views.py

```diff
@@ -1249,3 +1250,44 @@ def post(self, request):
                 {'error': str('Notification read request failed')},
                 status=HTTP_500_INTERNAL_SERVER_ERROR
             )
+
+
+class EnterpriseCustomerReportTypesView(APIView):
+    """
+    API for getting the report types associated with an enterprise customer
+    """
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated, permissions.IsAdminUser]
+    http_method_names = ['get']
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, enterprise_uuid: enterprise_uuid
+    )
+    def get(self, request, enterprise_uuid):
+        """
+        Get the dropdown choices for EnterpriseCustomerReportingConfiguration
+        """
+        enterprise_customer = get_enterprise_customer(enterprise_uuid)
+        if not enterprise_customer:
+            return Response({"detail": "Could not find the enterprise customer."}, status=HTTP_404_NOT_FOUND)
+
+        meta = models.EnterpriseCustomerReportingConfiguration._meta
+        pearson_data_type_choices = meta.get_field('data_type').choices
+        report_type_choices = meta.get_field('report_type').choices
+        delivery_method_choices = meta.get_field('delivery_method').choices
+        frequency_choices = meta.get_field('frequency').choices
+        non_pearson_data_type_choices = []
+        for data_type_choice in pearson_data_type_choices:
+            if data_type_choice[1] not in models.EnterpriseCustomerReportingConfiguration.PEARSON_ONLY_REPORTS:
+                non_pearson_data_type_choices.append(data_type_choice)
+        data_type_choices = (
+            pearson_data_type_choices if 'pearson' in enterprise_customer.slug else tuple(non_pearson_data_type_choices)
+        )
+        choices = {
+            'data_type': data_type_choices,
+            'report_type': report_type_choices,
+            'delivery_method': delivery_method_choices,
+            'frequency': frequency_choices
+        }
```

Comment:
can we make this method more generic, in the future, we don't have to add any new choices fields?

example code:
```
    meta = models.EnterpriseCustomerReportingConfiguration._meta

    choices = {}
    for field in meta.get_fields():
        if hasattr(field, 'choices'):
            choices[field.name]: field.choices

    if 'pearson' not in enterprise_customer.slug:
        # need to remove Pearson only reports
        all_data_types = choices.get('data_type', [])
        for index, data_type in all_data_types:
            if data_type[1] in models.EnterpriseCustomerReportingConfiguration.PEARSON_ONLY_REPORTS:
                del all_data_types[[index]
```